### ISSUE TITLE
False positive: Type narrowing not recognized after throw statement

### ISSUE THREAD

**Issue by janosh** [pheuter/svelte-check-rs#93]
When a variable is typed as \`T | undefined\` and there's a guard that throws if the value is undefined, svelte-check-rs doesn't recognize the type narrowing and still reports the variable as possibly undefined.

## Reproduction

```svelte
<script lang="ts">
  interface DataFile {
    figshare: string
    name: string
  }

  const data_files: Record<string, DataFile | undefined> = {}
  const mp_trj_data = data_files[\`mp_trj_json_gz\`]

  if (!mp_trj_data) {
    throw \`mp_trj_json_gz not found\`
  }
  // After throw, mp_trj_data should be narrowed to DataFile
</script>

<a href={mp_trj_data.figshare}>Link</a>
```

## Error

```
Error: 'mp_trj_data' is possibly 'undefined'. (ts(TS18048))
```

## Workaround

Use non-null assertion: \`mp_trj_data!.figshare\`

svelte-check-rs: 0.7.3
